### PR TITLE
Use `Lefthook` instead of `husky + lint-staged`

### PR DESCRIPTION
**Pull request by koralle**
See: https://github.com/yamada-ui/yamada-ui/pull/2339

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,11 +56,11 @@
...
-    "husky": "^9.0.11",
-    "lint-staged": "^15.2.7",
+    "lefthook": "^1.7.7",
     "prettier": "^3.3.2",
     "rimraf": "^5.0.9",
     "tsx": "^4.16.2",
     "typescript": "^5.5.3"
-  }
+  },
+  "packageManager": "pnpm@8.15.6+sha256.01c01eeb990e379b31ef19c03e9d06a14afa5250b82e81303f88721c99ff2e6f"
```
Comment (hirotomoyamada) [inline package.json:65]: Why is this line necessary?

Comment (koralle) [reply package.json]: removed.

**Review by setodeve — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]
